File: GlideExample/app/src/main/java/kr/teamcadi/glideexample/MainActivity.java
package kr.teamcadi.glideexample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ImageView;

import com.bumptech.glide.Glide;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ImageView ivGlide = (ImageView) findViewById(R.id.iv_glide);

        /*
        ** glide 는 3가지 메소드를 필수적 사용
        ** 1. with : Context -> this
        ** 2. load : 이미지를 받아올 경로  외부에서 호출해서 받아옴(인터넷 권한 활성화)
        ** 3. into : 받아와서 어디에 넣을 건지
         */

        //sns에 있는 사진을 가져오려면 load안에 이미지 주소 복사를 한 후 bit.ly에 들어와서 링크 길이를 줄인다!!! -> load메소드 안에 넣음
        Glide.with(this).load("http://goo.gl/gEgYUd").into(ivGlide);
    }
}

File: MvvmPractice/app/src/main/java/kr/teamcadi/mvvmpractice/MainActivity.kt
package kr.teamcadi.mvvmpractice

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.LinearLayout
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import kr.teamcadi.mvvmpractice.databinding.ActivityMainBinding
import kr.teamcadi.mvvmpractice.db.SubscriberDAO
import kr.teamcadi.mvvmpractice.db.SubscriberDatabase
import kr.teamcadi.mvvmpractice.db.SubscriberRepository

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var subscriberViewModel: SubscriberViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this,R.layout.activity_main)
        val dao:SubscriberDAO = SubscriberDatabase.getInstance(application).subscriberDAO
        val repository = SubscriberRepository(dao)
        val factory = SubscriberViewModelFactory(repository)
        subscriberViewModel = ViewModelProvider(this,factory).get(SubscriberViewModel::class.java)
        binding.myViewModel = subscriberViewModel

        binding.lifecycleOwner = this
        initRecyclerView()
    }

    private fun initRecyclerView(){
        binding.subscriberRecyclerView.layoutManager = LinearLayoutManager(this)
        displaySubscribersList()

    }

    private fun displaySubscribersList() {
        subscriberViewModel.subscribers.observe(this, Observer { Log.i("MYTAG",it.toString())
            binding.subscriberRecyclerView.adapter = MyRecyclerViewAdapter(it)
        })
    }
}